### ISSUE TITLE
etcd defrag, compaction and endpoint status

### ISSUE THREAD

**Issue by jsalgado78** [k0sproject/k0s#1920]
### Is your feature request related to a problem? Please describe.

It would be interesting to have etcdctl commands like defrag, compaction and endpoint status in k0s etcd subcommand for etcd database maintenance and monitoring

**Comment by soider** [comment]
I might suggest you to use /var/lib/k0s/pki/apiserver-etcd-client.crt certififacte to got an access for the embedded etcd server

**Comment by jsalgado78** [comment]
Thanks. I use etcdctl for etcd maintenance but k0s etcd subcommand is very limited and it would be interesting to implement subcommands for etcd database maintenance

**Comment by till** [comment]
I agree... the etcd is currently a blackbox and when it fails it's hard to get logs from the journal that belong to it. But to make it work in the meantime:

```sh
sudo etcdctl defrag --command-timeout=120s --cert=/var/lib/k0s/pki/apiserver-etcd-client.crt --key=/var/lib/k0s/pki/apiserver-etcd-client.key --insecure-skip-tls-verify=true
```

**Comment by jnummelin** [comment]
Not planned currently, community PRs welcome 😄
